File: jdroid-android-firebase-remoteconfig/src/main/java/com/jdroid/android/firebase/remoteconfig/FirebaseRemoteConfigFetchCommand.java
package com.jdroid.android.firebase.remoteconfig;

import android.content.Context;
import android.os.Bundle;

import com.jdroid.android.firebase.jobdispatcher.ServiceCommand;

public class FirebaseRemoteConfigFetchCommand extends ServiceCommand {

	private static final String SET_EXPERIMENT_USER_PROPERTY = "setExperimentUserProperty";

	public void start() {
		start( false);
	}

	public void start(Boolean setExperimentUserProperty) {
		Bundle bundle = new Bundle();
		bundle.putBoolean(FirebaseRemoteConfigFetchCommand.SET_EXPERIMENT_USER_PROPERTY, setExperimentUserProperty);
		start(bundle);
	}

	@Override
	protected boolean execute(Context context, Bundle bundle) {
		Boolean setExperimentUserProperty = bundle.getBoolean(SET_EXPERIMENT_USER_PROPERTY);
		FirebaseRemoteConfigLoader.get().fetch(setExperimentUserProperty, null);
		return false;
	}
}

File: jdroid-android-firebase-remoteconfig/src/main/java/com/jdroid/android/firebase/remoteconfig/FirebaseRemoteConfigFetchCommand.kt
package com.jdroid.android.firebase.remoteconfig

import android.content.Context
import android.os.Bundle

import com.jdroid.android.firebase.jobdispatcher.ServiceCommand

class FirebaseRemoteConfigFetchCommand : ServiceCommand() {

    companion object {
        private const val SET_EXPERIMENT_USER_PROPERTY = "setExperimentUserProperty"
    }

    override fun start() {
        start(false)
    }

    fun start(setExperimentUserProperty: Boolean?) {
        val bundle = Bundle()
        bundle.putBoolean(FirebaseRemoteConfigFetchCommand.SET_EXPERIMENT_USER_PROPERTY, setExperimentUserProperty!!)
        start(bundle)
    }

    override fun execute(context: Context, bundle: Bundle): Boolean {
        val setExperimentUserProperty = bundle.getBoolean(SET_EXPERIMENT_USER_PROPERTY)
        FirebaseRemoteConfigLoader.get().fetch(setExperimentUserProperty, null)
        return false
    }
}
